src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -705,7 +705,7 @@ internal static bool MatchesModulePath(string modulePath, string requiredPath)
 #if UNIX
             StringComparison strcmp = StringComparison.Ordinal;
 #else
-            StringComparison strcmp = StringComparison.OrdinalIgnoreCase;
+            const StringComparison strcmp = StringComparison.OrdinalIgnoreCase;
```

Comment:
Unix case should be fixed too.

Reply:
`StrCmp`? or maybe `IgnoreCaseOpt`?

Reply:
`strCmp` is probably a good option

Reply:
Our code convention is to start with upper case for const.

Reply:
My thinking is that that's only for global constants, rather than local constant variables

Reply:
We should follow CodeFactor and Code Analyzers.

Reply:
I am persuaded by @rjmholt's reasoning but our [coding guidelines](https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/coding-guidelines.md#naming-conventions) state to: "Use `PascalCase to name constant local variables and fields." and is common with [dotnet coding style](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md).

On the other hand, the style of the code in `System.Management.Automation` appears to use mostly `camelCase` over `PascalCase` for local constants and we should generally follow the style of the surrounding code.

`strCmp` would be a violation of [SA1303ConstFieldNamesMustBeginWithUpperCaseLetter](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1303.md) but there is an [open issue](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2082) as to whether the rule should apply to local constants as well as fields.

I think there could be a seperate discussion on changes to the naming conventions so I am minded not to make this change as part of this PR.

Reply:
My vote would be for keeping the current variable name unchanged in all cases, and just adding the `const` modifier to the declaration. We can decide later on how local constants should be cased

Reply:
Issue opened for local constants naming convention: #13247.